src/vos/vos_obj.c

```diff
@@ -201,12 +283,17 @@ update_read_timestamps(struct vos_ts_set *ts_set, daos_epoch_t epoch,
 		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
 		return;
 	}
+	if (entry->te_punch_propagated)
+		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
 	entry = vos_ts_set_get_entry_type(ts_set, VOS_TS_TYPE_DKEY, 0);
 	entry->te_ts_rh = MAX(entry->te_ts_rh, epoch);
 	if (ts_set->ts_init_count == 3) {
 		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
 		return;
 	}
+	if (entry->te_punch_propagated)
+		entry->te_ts_rl = MAX(entry->te_ts_rl, epoch);
```

Comment:
(style)  'te' may be misspelled - perhaps 'the'?
(style)  'te' may be misspelled - perhaps 'the'?